src/mixins/PageData.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable max-classes-per-file */
+import { Component, Vue } from 'vue-property-decorator';
+import { Route, RouteCallback } from 'vue-router';
+
+/**
+ * Abstraction for handling fetching data for a page while allowing it to be cached
+ * if provided by SSR
+ *
+ * Here be abstract class mixin hackery
+ * https://github.com/vuejs/vue-class-component/issues/91#issuecomment-451422547
+ */
+// @ts-ignore
+@Component
+abstract class PageDataMixinAbstract extends Vue {
+  abstract async fetchPageData(): Promise<any>;
+
+  private async fetchPageDataInternal() {
+    this.$vxm.pageData.data = await this.fetchPageData();
+  }
+
+  async serverPrefetch() {
+    await this.fetchPageDataInternal();
+  }
+
+  mounted() {
+    if (this.$vxm.pageData.refetch || this.$vxm.pageData.data == null) {
```

Comment:
Any specific reasons for not using `===`?
If not, I'm adding that as a rule (And we can always use `!(this.$vxm.pageData.data)`)

Reply:
No, I'm surprised the linter didn't catch that (there might be some specific exclusion for null due to the way it works?)